signer/local/local.go

```diff
@@ -163,7 +163,9 @@ func PopulateSubjectFromCSR(s *signer.Subject, req pkix.Name) pkix.Name {
 	replaceSliceIfEmpty(&name.Locality, &req.Locality)
 	replaceSliceIfEmpty(&name.Organization, &req.Organization)
 	replaceSliceIfEmpty(&name.OrganizationalUnit, &req.OrganizationalUnit)
-
+	if name.SerialNumber == "" {
```

Comment:
Nope, you've got it flipped. This is for if name doesn't have the SerialNumber set already. (e.g. boulder or another client didn't pass one in to the Subject)